src/dcegm/fast_upper_envelope.py

```diff
@@ -825,57 +668,555 @@ def _backward_scan(
     )
 
 
-def _evaluate_point_on_line(
-    x1: float, y1: float, x2: float, y2: float, point_to_evaluate: float
-) -> float:
-    """Evaluate a point on a line.
+def update_bools_and_idx_to_inspect(idx_to_inspect, update_idx, case_2, case_5):
+    """Update indicators and index of the point to be inspected in the next period.
+
+    The indicators are booleans that capture cases where
+    - we have saved the last point we checked already, and
+    - the last point we inspected is an intersection point.
 
     Args:
-        x1 (float): x coordinate of the first point.
-        y1 (float): y coordinate of the first point.
-        x2 (float): x coordinate of the second point.
-        y2 (float): y coordinate of the second point.
-        point_to_evaluate (float): The point to evaluate.
+        idx_to_inspect (int): Index of the point to be inspected in the current
+            iteration.
+        update_idx (bool): Indicator if the index should be updated.
+        case_2 (bool): Indicator if we have reached the final point on the grid.
+        case_5 (bool): Indicator if we are in the situation where we added the
```

Comment:
case_5 was a little unclear to me. This one should be updated.

Reply:
It will. But at a later point!